File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/WeakHandler.java
package me.ycdev.android.lib.common.utils;

import java.lang.ref.WeakReference;

import android.os.Handler;
import android.os.Message;
import androidx.annotation.NonNull;

@SuppressWarnings({"unused", "WeakerAccess"})
public class WeakHandler extends Handler {
    private WeakReference<Handler.Callback> mTargetHandler;

    public WeakHandler(@NonNull Handler.Callback msgHandler) {
        mTargetHandler = new WeakReference<>(msgHandler);
    }

    @Override
    public void handleMessage(Message msg) {
        Handler.Callback realHandler = mTargetHandler.get();
        if (realHandler != null) {
            realHandler.handleMessage(msg);
        }
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/WeakHandler.kt
package me.ycdev.android.lib.common.utils

import android.os.Handler
import android.os.Looper
import android.os.Message
import java.lang.ref.WeakReference

@Suppress("unused")
class WeakHandler(looper: Looper, msgHandler: Callback) : Handler(looper) {
    private val targetHandler: WeakReference<Callback> = WeakReference(msgHandler)

    constructor(msgHandler: Callback) : this(Looper.myLooper()!!, msgHandler)

    override fun handleMessage(msg: Message) {
        val realHandler = targetHandler.get()
        realHandler?.handleMessage(msg)
    }
}
